src/js/yeldaWebchatInjector.js

```diff
@@ -775,38 +775,105 @@ class YeldaChat {
     return isFound
   }
 
+  /**
+   * Get the app env from the assistant url
+   * @param {String} assistantUrl 
+   * @returns {String} env
+   */
+  getAppEnv(assistantUrl) {
+    if (config.REGEX_HOST.STAGING.test(assistantUrl)) {
+      return config.APP_ENV_VALUES.STAGING
+    }
+
+    // By default, use the production env
+    return config.APP_ENV_VALUES.PRODUCTION
+  }
+
+  /**
+   * Get the chat bubble url base based on isFallback
+   * @param {Object} data { assistantUrl, assistantId, settingId }
+   * @param {Boolean} isFallback fallback means yelda endpoint
+   * @returns {String} url base
+   */
+  getChatBubbleUrlBase(data, isFallback) {
+    if (isFallback) {
+      return `${data.assistantUrl}/assistants`
+    }
+    return config.CHAT_BUBBLE_EXTERNAL_ENDPOINT
+  }
+
   /**
    * return getAssistantSettings endpoint URL from data
    * @param {Object} data { assistantUrl, assistantId, settingId }
+   * @param {String} urlBase
    * @return {String} url
    */
-  getAssistantSettingsUrl(data) {
+  getAssistantSettingsUrl(data, isFallback) {
+    const urlBase = this.getChatBubbleUrlBase(data, isFallback)
+    // The app env is required as query for the external chatbubble endpoint
+    // If the env was not explicitely set in data, guess it from the assistant url
+    const appEnv = data.env || this.getAppEnv(data.assistantUrl)
+    const envQuery = `env=${appEnv}`
     if(data.settingId) {
-      return  `${data.assistantUrl}/assistants/settings/${data.settingId}/chatBubble`
+      return  `${urlBase}/settings/${data.settingId}/chatBubble?${envQuery}`
     }
 
-    return  `${data.assistantUrl}/assistants/${data.assistantId}/chatBubble/${data.locale}`
+    return  `${urlBase}/${data.assistantId}/chatBubble/${data.locale}?${envQuery}`
```

Comment:
:question: I think it's missing in the route ?
```js
      {
        method: 'GET',
        path: '/assistants/{assistantId}/chatBubble/{locale?}',
        config: {
          auth: false,
          handler: async (request, h) => {
            return AssistantController.getAssistantChatBubbleSetting(request, h)
          },
          description: 'Return assistant chat bubble if any',
          tags: ['api', 'assistant', 'media'],
          validate: {
            params: {
              assistantId: idValidation,
              locale: Joi.string()
                .valid(...configDefault.PLATFORMS_AVAILABLE_LOCALES)
                .optional()
            }
          }
        }

Reply:
What is missing? Do you mean the `env` query? 
If so we don't need specify it in the Joi validation. Joi just ignores it if not specified in the validate

Reply:
https://staging.yelda.ai/assistants/5b7edb2b1060312cfeaa7932/chatBubble/fr_FR?env=staging => indeed

but https://staging.yelda.ai/assistants/5b7edb2b1060312cfeaa7932/chatBubble/fr_FR?env=production does work too 

Did we not say before that we should check for the env ? I don't see it in `getAssistantChatBubbleSetting` :thinking:

Reply:
For the new endpoint, we needed the `env` to format the Redis key.

But should the `env` from the query overwrite the NODE_ENV in yelda? 
In Yelda we currently don't use the `env` to format the redis key because Redis is already initialised with the NODE_ENV as prefix.

Or maybe you meant that we should compare the `env` from the query with the NODE_ENV, if they are not the same, we don't return anything?

Reply:
> I understood that the env was just for the new endpoint, not the old one, sorry

i am sorry, you're right, i mixed it !

> Or maybe you meant that we should compare the env from the query with the NODE_ENV, if they are not the same, we don't return anything?

when i wrote the feedback (with incorrect idea in mind, i have been thinking to that ... but let's clarify monday with a clear mind)
The point is, i would prefer to avoid mixing staging / prod / local data in redis (even if it's not likely to happen)


Another danger i just though about is if someone is connected to our redis but with a local copy of the DB and change something, we could have incorrect data in redis => i might create a special redis for prod :thinking:  i'll open an issue for me